com.unity.ml-agents.extensions/Runtime/Input/InputActionActuator.cs

```diff
@@ -0,0 +1,108 @@
+#if MLA_INPUT_SYSTEM
+
+#define ACTUATOR_DEBUG
+#undef ACTUATOR_DEBUG
+using Unity.MLAgents.Actuators;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+
+#if ACTUATOR_DEBUG
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem.Controls;
+using UnityEngine.InputSystem.LowLevel;
+#endif
+
+namespace Unity.MLAgents.Extensions.Runtime.Input
+{
+    public class InputActionActuator : IActuator, IHeuristicProvider
+    {
+        bool m_IsInHeuristicMode;
+        InputAction m_Action;
+        IRLActionInputAdaptor m_InputAdaptor;
+        InputDevice m_Device;
+        InputControl m_Control;
+        Agent m_Agent;
+
+#if ACTUATOR_DEBUG
+        float m_Time;
+        bool m_Flip;
+#endif
+
+        public InputActionActuator(Agent agent,
+            InputAction action,
+            IRLActionInputAdaptor adaptor,
+            bool isInHeuristicMode)
+        {
+            m_Agent = agent;
+            m_IsInHeuristicMode = isInHeuristicMode;
+            Name = $"InputActionActuator-{action.name}";
+            m_Action = action;
+            m_InputAdaptor = adaptor;
+            ActionSpec = adaptor.GetActionSpecForInputAction(m_Action);
+        }
+
+        internal void SetDevice(InputDevice device)
+        {
+            m_Device = device;
+            m_Control = m_Device.GetChildControl(m_Action.name);
+        }
+
+        public void OnActionReceived(ActionBuffers actionBuffers)
+        {
+            if (!m_IsInHeuristicMode)
+            {
+                m_InputAdaptor.QueueInputEventForAction(m_Action, m_Control, ActionSpec, actionBuffers);
+            }
+            else
+            {
+                m_Agent.OnActionReceived(actionBuffers);
```

Comment:
I don't think we want to do anything for the Agent here, right? The buffer is only sized for this actuator, which is likely a different ActionSpec from the Agent.

Reply:
I think (although now we should double check) that the full actionBuffer will get recorded for demonstrations, not just the one that gets passed to Agent.OnActionReceived.

Reply:
yeah, this should be removed.  I'll double check the demo recording.  I think it was getting the buffer from the actuatorManager

Reply:
the AgentInfo now has a reference to the ActionBuffers and copies the action from the brain in decide action.  So it looks like it's hooked up to get the complete action buffer.